homeassistant/components/moon/config_flow.py

```diff
@@ -0,0 +1,37 @@
+"""Config flow to configure the Moon integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_NAME, DOMAIN
+
+
+class MoonConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for Moon."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=user_input.get(CONF_NAME, DEFAULT_NAME),
+                data={},
+            )
+
+        return self.async_show_form(step_id="user", data_schema=vol.Schema({}))
```

Comment:
Can we remove the data schema as there's no user data input needed?

Reply:
Currently not.
It's typed as required.

Reply:
Ok, we should fix the typing and make it optional here:
https://github.com/home-assistant/core/blob/624c3be3aed79f1d8321169abb5cb85098768160/homeassistant/data_entry_flow.py#L411

Reply:
https://github.com/home-assistant/core/pull/67637